### PR TITLE
[Win32] Make basic system checks use psutil & win32pdh rather than WMI

### PR DESCRIPTION
**Pull request by derekwbrown**
### What does this PR do?

Makes the basic system checks use other means besides WMI.  The WMI checks are having anecdotal reports of slow checks and other problems.  This change removes use of the WMI check (though leaves the WMI checks available, since standalone WMI checks are supported).

### Motivation

Support tickets.

### Additional Notes

While there are reports of the WMI check failing in unpredictable ways, it's not been reported in-house.  Therefore, there's no conclusive proof this checkin is an improvement.

### CODE REVIEW

File: checks/libs/win/winpdh.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
+import time
+import sys
+import win32pdh
+import psutil
+
+class WinPDHCounter(object):
+    
+    def __init__(self, class_name, instance_name):
+        self._class_name = class_name
+        self._instance_name = instance_name
```
Comment (gmmeyer) [inline checks/libs/win/winpdh.py]: Could we make this class stateless, make the functions into class methods and just mass these variables into those methods? If I'm not missing anything, it seems like you're only calling it once with each of these.

Comment (derekwbrown) [inline checks/libs/win/winpdh.py]: I did it this way to match as closely as possible the previous calling convention.  I'd be happy to make that change if it's helpful.

File: checks/libs/win/winpdh.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+    def get_single_value(self):
+        hq = win32pdh.OpenQuery()
+        path = win32pdh.MakeCounterPath((None, self._class_name, None, None, 0, self._instance_name))
+        counter_handle = win32pdh.AddCounter(hq, path)
+        win32pdh.CollectQueryData(hq)
+        t, val = win32pdh.GetFormattedCounterValue(counter_handle, win32pdh.PDH_FMT_LARGE)
```
Comment (yannmh) [inline checks/libs/win/winpdh.py]: `PDH_FMT_LARGE` vs `PDH_FMT_LONG ` (in `get_all_values`). Also https://www.cac.cornell.edu/wiki/index.php?title=Performance_Data_Helper_in_Python_with_win32pdh suggests to use `PDH_FMT_FLOAT`

File: checks/libs/win/winpdh.py [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    
+    def __init__(self, class_name, instance_name):
+        self._class_name = class_name
+        self._instance_name = instance_name
+
+    def get_single_value(self):
+        hq = win32pdh.OpenQuery()
+        path = win32pdh.MakeCounterPath((None, self._class_name, None, None, 0, self._instance_name))
+        counter_handle = win32pdh.AddCounter(hq, path)
+        win32pdh.CollectQueryData(hq)
```
Comment (yannmh) [inline checks/libs/win/winpdh.py]: From what i understand `hq` is stateful

> Some performance counters are instantaneous. Others require you collect data for a few seconds. See references for details. A safe bet is to collect data twice before getting values.
 
(https://www.cac.cornell.edu/wiki/index.php?title=Performance_Data_Helper_in_Python_with_win32pdh)


Shall it be kept in the instance to persist over time?

File: checks/libs/win/winpdh.py [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    
+            win32pdh.CollectQueryData(hq)
+            try:
+                t, val = win32pdh.GetFormattedCounterValue(counter_handle, win32pdh.PDH_FMT_LONG)
+                ret[inst] = val
+            except Exception as e:
+                # exception usually means self type needs two data points to calculate. Wait
+                # a bit and try again
+                time.sleep(0.01)
+                win32pdh.CollectQueryData(hq)
```
Comment (yannmh) [inline checks/libs/win/winpdh.py]: Same as above about persisting query objects. What do you think?

File: checks/libs/win/winpdh.py [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+            counter_handle = win32pdh.AddCounter(hq, path)
+    
+            win32pdh.CollectQueryData(hq)
+            try:
+                t, val = win32pdh.GetFormattedCounterValue(counter_handle, win32pdh.PDH_FMT_LONG)
+                ret[inst] = val
+            except Exception as e:
+                # exception usually means self type needs two data points to calculate. Wait
+                # a bit and try again
+                time.sleep(0.01)
```
Comment (yannmh) [inline checks/libs/win/winpdh.py]: Let's put this time in a CONSTANT at the class or module level.

File: checks/libs/win/winpdh.py [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+                win32pdh.CollectQueryData(hq)
+                # if we get exception self time, just return it up
+                try:
+                    t, val = win32pdh.GetFormattedCounterValue(counter_handle, win32pdh.PDH_FMT_LONG)
+                    ret[inst] = val
+                except Exception as e:
+                    win32pdh.CloseQuery(hq)
+                    raise e
+            win32pdh.CloseQuery(hq)
```
Comment (yannmh) [inline checks/libs/win/winpdh.py]: Python has a `finally` clause for that 🍬 

```python
try:
    try:
    except Exception:
        do_something()
except Exception:
    do_something()
finally:
    win32pdh.CloseQuery(hq)  # Where we close the query

```

File: checks/libs/win/winpdh.py (hunk 3 above)
Comment (derekwbrown) [inline checks/libs/win/winpdh.py]: see below.

File: checks/libs/win/winpdh.py (hunk 4 above)
Comment (derekwbrown) [inline checks/libs/win/winpdh.py]: So, this was taken directly from a sample on how to use winpdh.  I don't know if it's better to just to the brief sleep, or do the state as you mention above.

**Comment by derekwbrown** [comment]
Older PR, got tripped up with git merging.  Will create new PR.

**Pull request closed without merging**